File: MyCalendar.java
import java.util.ArrayList;
import java.util.Comparator;
import java.util.List;

public class MyCalendar {
    List<int[]> bookedSlots;

    public MyCalendar(){
        bookedSlots = new ArrayList<>();
    }

    public boolean isFree(int start, int end){
        // inside
        for (int[] slot : bookedSlots){
            if(slot[0] <= start && slot[1] >= end){
                return false;
            }
            if(slot[0] <= start && slot[1] >=start){
                return false;
            }
            if(slot[0] <= end && slot[1] >= end){
                return false;
            }
        }
        return true;
    }

    public boolean bookSlot(int start, int end){
        if(!isFree(start, end)){
            return false;
        }
        bookedSlots.add(new int[]{start, end});
        bookedSlots.sort(Comparator.comparingInt(it -> it[0]));
        return true;
    }
}

File: TaxiService/MyCalendar.cs
using System;
using System.Collections.Generic;

public class MyCalendar
{
    private List<int[]> bookedSlots;

    public MyCalendar()
    {
        bookedSlots = new List<int[]>();
    }

    public bool IsFree(int start, int end)
    {
        foreach (var slot in bookedSlots)
        {
            // Check if there's an overlap
            if (slot[0] <= start && slot[1] >= end)
            {
                return false;
            }
            if (slot[0] <= start && slot[1] >= start)
            {
                return false;
            }
            if (slot[0] <= end && slot[1] >= end)
            {
                return false;
            }
        }
        return true;
    }

    public bool BookSlot(int start, int end)
    {
        if (!IsFree(start, end))
        {
            return false;
        }
        bookedSlots.Add(new int[] { start, end });
        bookedSlots.Sort((a, b) => a[0].CompareTo(b[0]));
        return true;
    }
}